src/camera/camera_simple_radial.h

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 ETH Zürich, Thomas Schöps
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+//    this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+
+#pragma once
+
+#include <math.h>
+
+#include <Eigen/Core>
+#include <glog/logging.h>
+#include "camera/camera_base.h"
+#include "camera/camera_base_impl.h"
+#include "camera/camera_base_impl_radial.h"
+
+namespace camera {
+
+// Models pinhole cameras with a polynomial distortion model.
+class SimpleRadialCamera : public RadialBase<SimpleRadialCamera> {
+ public:
+  SimpleRadialCamera(int width, int height, float f,
+                     float cx, float cy, float k);
+
+  SimpleRadialCamera(int width, int height, const float* parameters);
+
+  static constexpr int ParameterCount() {
+    return 3 + 1;
+  }
+
+  static constexpr bool UniqueFocalLength() {
+    return true;
+  }
+
+  void InitCutoff();
+
+  inline float DistortionFactor(const float r2) const {
+    return 1.0f + r2 * k1_;
+  }
+  // Returns the derivatives of the image coordinates with respect to the
```

Comment:
Again, this might apply when calling another function that in turn calls this one, but it does not apply to this function itself, does it?